src/Sarif.Driver/Sdk/EntryPointUtilities.cs

```diff
@@ -0,0 +1,111 @@
+﻿// Copyright (c) Microsoft. All rights reserved.
+// Licensed under the MIT license. See LICENSE file in the project root for full license information.
+
+using System;
+using System.Collections.Generic;
+using System.IO;
+
+namespace Microsoft.CodeAnalysis.Sarif.Driver.Sdk
+{
+    public static class EntryPointUtilities
+    {
+        internal static string[] GenerateArguments(
+            string[] args,
+            IFileSystem fileSystem,
+            IEnvironmentVariables environmentVariables)
+        {
+            List<string> expandedArguments = new List<string>();
+
+            foreach (string argument in args)
+            {
+                if (argument[0] != '@' &&
+                    (argument.Length == 1 || argument[1] != '@'))
+                {
+                    expandedArguments.Add(argument);
+                    continue;
+                }
+
+                string responseFile = argument.Trim('"').Substring(1);
```

Comment:
>  [](start = 0, length = 1)

Consider breaking some of this logic into helper functions, so that this method reduces to (for example):
```
foreach (string argument in args)
{
    if (IsResponseFile(argument))
    {
        AddResponseFileContents(ResponseFileName(argument), expandedArguments);
    }
    else
    {
        argument = UnescapeAtSign(argument);
        expandedArguments.Add(argument);
    }
}
```